app/src/display/widgets/wpm_status.c

```diff
@@ -31,37 +36,42 @@ void wpm_status_init() {
     lv_style_set_text_line_space(&label_style, LV_STATE_DEFAULT, 1);
 }
 
-void set_wpm_symbol(lv_obj_t *label, int wpm) {
+struct wpm_status_state wpm_status_get_state(const zmk_event_t *eh) {
+    return (struct wpm_status_state){.wpm = zmk_wpm_get_state()};
+};
+
+void set_wpm_symbol(lv_obj_t *label, struct wpm_status_state state) {
     char text[4] = {};
 
-    LOG_DBG("WPM changed to %i", wpm);
-    sprintf(text, "%i ", wpm);
+    LOG_DBG("WPM changed to %i", state.wpm);
+    snprintf(text, sizeof(text), "%i ", state.wpm);
 
     lv_label_set_text(label, text);
 }
 
+void wpm_status_update_cb(struct wpm_status_state state) {
+    struct zmk_widget_wpm_status *widget;
+    SYS_SLIST_FOR_EACH_CONTAINER(&widgets, widget, node) { set_wpm_symbol(widget->obj, state); }
+}
+
+ZMK_DISPLAY_WIDGET_LISTENER(widget_wpm_status, struct wpm_status_state, wpm_status_update_cb,
+                            wpm_status_get_state)
+ZMK_SUBSCRIPTION(widget_wpm_status, zmk_wpm_state_changed);
+
 int zmk_widget_wpm_status_init(struct zmk_widget_wpm_status *widget, lv_obj_t *parent) {
     wpm_status_init();
+    // widget_wpm_status_refresh_state(NULL);
```

Comment:
Are these intended to be left commented out?

Reply:
Nope, removed.